src/main/java/org/gelecekbilimde/scienceplatform/auth/controller/RoleSelfApplicationController.java

```diff
@@ -0,0 +1,77 @@
+package org.gelecekbilimde.scienceplatform.auth.controller;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.gelecekbilimde.scienceplatform.auth.model.RoleApplication;
+import org.gelecekbilimde.scienceplatform.auth.model.mapper.RoleApplicationToRoleSelfApplicationsResponseMapper;
+import org.gelecekbilimde.scienceplatform.auth.model.request.RoleSelfApplicationListRequest;
+import org.gelecekbilimde.scienceplatform.auth.model.response.RoleSelfApplicationsResponse;
+import org.gelecekbilimde.scienceplatform.auth.service.RoleSelfApplicationService;
+import org.gelecekbilimde.scienceplatform.common.model.BasePage;
+import org.gelecekbilimde.scienceplatform.common.model.response.PagingResponse;
+import org.gelecekbilimde.scienceplatform.common.model.response.SuccessResponse;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Validated
+@RestController
+@RequestMapping("/api/v1")
+@RequiredArgsConstructor
+class RoleSelfApplicationController {
+
+	private final RoleSelfApplicationService roleSelfApplicationService;
+
+
+	private final RoleApplicationToRoleSelfApplicationsResponseMapper roleApplicationToRoleSelfApplicationsResponseMapper = RoleApplicationToRoleSelfApplicationsResponseMapper.initialize();
```

Comment:
@coderabbitai The area here cannot be injected by spring. Because it's not a bean. In addition, it cannot inject because we use lombok anotations. Don't make this suggestion for mappers.